### PR TITLE
Feat filter more stories

### PR DESCRIPTION
**Pull request by anitavincent**
#### What is the purpose of this pull request?
Add more stories and add documentation for new composable filter components

#### How should this be manually tested?
Check docs under experimental/filter

check new async stories

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui-docs/content/filter/code.mdx [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+
+</Section>
+
+<Section>
+
+### useFilterState and useFilterMultipleState props
+
+| Name           | Type                                           | Description                                                                      | Required | Default  |
+| -------------- | ---------------------------------------------- | -------------------------------------------------------------------------------- | -------- | -------- |
+| fullList       | {id: string, label: string, value?: AnyObject} | Full option list that will be used for searching. Only use it if you need search.| 🚫       | -        |
```
Comment (marcelovicentegc) [inline packages/admin-ui-docs/content/filter/code.mdx:78]:
```suggestion
| fullList       | {id: string, label: string, value?: AnyObject}[] | Full option list that will be used for searching. Only use it if you need search.| 🚫       | -        |
```

I'm assuming this is an array, is that right?

File: packages/admin-ui-docs/content/filter/code.mdx [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+
+interface FilterMultipleState<T> extends GenericFilterState {
+  /**
+   * State that handles rendered options and search
+   */
+  combobox: ComboboxMultipleState<FilterOption<T>>
+  /**
+   * Options that are currently applied
+   */
+  appliedItems: ItemList<T>
```
Comment (marcelovicentegc) [inline packages/admin-ui-docs/content/filter/code.mdx:139]: Would it be relevant to showcase these types as well? My only concern would be the documentation getting too long unnecessarily

File: packages/admin-ui-docs/content/filter/code.mdx [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+## Filter
+
+You can build a simple single select filter with the disclosure, popover, listbox, footer and optionRadio components.
+
+```jsx
+export function Filter() {
+  const filterState = useFilterState()
+
+    <>
+  return (
```
Comment (marcelovicentegc) [inline packages/admin-ui-docs/content/filter/code.mdx:189]:
```suggestion
  return (
       <>
```

File: packages/admin-ui-docs/content/filter/code.mdx [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+}
+```
+
+## FilterSearch
+
+To add search to the basic filter you must add the searchbox component and also 
+start interacting with the combobox state returned by the hook. 
+
+To use the components native search you must provide the full list of items, 
+when the user performs a search operation `state.combobox.matches` will
```
Comment (marcelovicentegc) [inline packages/admin-ui-docs/content/filter/code.mdx:237]:
```suggestion
To use the components native search you must provide the full list of items. 
When the user performs a search operation, `state.combobox.matches` will 
```

File: packages/admin-ui-docs/content/filter/overview.mdx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+### Anatomy
+
+There are four types of filters that can be composed with admin-ui's filter 
+components. They are implemented by nesting the composable filter components 
+to form specific structures.
+
+Basic filter composable component anatomy
+
+
```
Comment (marcelovicentegc) [inline packages/admin-ui-docs/content/filter/overview.mdx:22]:
```suggestion
```

Comment (marcelovicentegc) [reply packages/admin-ui-docs/content/filter/overview.mdx:22]: I guess there's an extra line here 😄

Comment (marcelovicentegc) [reply packages/admin-ui-docs/content/filter/overview.mdx:22]: Or maybe that was intentional (seeing the lines below follow the same pattern)

File: packages/admin-ui-docs/content/filter/overview.mdx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+|__ FilterDisclosure
+|__ FilterPopover
+    |__ FilterSearchbox
+    |__ FilterListbox
+    |   |__ . . .  
+    |__ FilterFooter
+```
+
+Searchable filter with multiple option selection is just an extension of the multiple filter
+
```
Comment (marcelovicentegc) [inline packages/admin-ui-docs/content/filter/overview.mdx:61]:
```suggestion
```

Maybe this is not a pattern after all 😂? Let me know!

**Review by marcelovicentegc — APPROVED** [review]
Just left a couple of suggestions, but LGTM! Excellent doc! 

Just one tip, Anita: what do you think of adding a `Composition` section under `overview`, and moving the `Anatomy` section to the code tab? I know that the `PageHeader` and the `Table` components are following this pattern.

File: packages/admin-ui-docs/content/filter/overview.mdx (hunk 6 above)
Comment (anitavincent) [reply packages/admin-ui-docs/content/filter/overview.mdx:59]: Don't know why that was there

File: packages/admin-ui-docs/content/filter/code.mdx [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+### useFilterState and useFilterMultipleState props
+
+| Name           | Type                                           | Description                                                                      | Required | Default  |
+| -------------- | ---------------------------------------------- | -------------------------------------------------------------------------------- | -------- | -------- |
+| fullList       | {id: string, label: string, value?: AnyObject} | Full option list that will be used for searching. Only use it if you need search.| 🚫       | -        |
+
+
+both filters return an object that extends the following type
+
+```ts
```
Comment (lucasaarcoverde) [inline packages/admin-ui-docs/content/filter/code.mdx:83]: What do you think of adding those types as Prop tables? You can name the prop table section title with the type name

File: packages/admin-ui-docs/content/filter/code.mdx [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+          <FilterOptionRadio id="#2" label="Empty" />
+          <FilterOptionRadio id="#3" label="Half full" />
+        </FilterListbox>
+        <FilterFooter />
+      </FilterPopover>
+    </>
+  )
+}
+```
+
```
Comment (lucasaarcoverde) [inline packages/admin-ui-docs/content/filter/code.mdx:205]: I think you forgot to close the Section tags, from this line til the end
```suggestion
</Section>

<Section hasExplanation>

...

</Section>
```

File: packages/admin-ui-docs/content/filter/code.mdx [hunk 9]
```diff
@@ -0,0 +1,305 @@
...
+| FilterOptionRadio     |  Radio option component, this should be used only on single selection filters      |
+| FilterSearchbox       |  Search box component that should be used only when building searchable filters    |
+| FilterGroup           |  A grouping component that display the disclosures as cohesive bar                 |
+
+
+</Section>
+
+<Section hasExplanation>
+
+## Filter
```
Comment (lucasaarcoverde) [inline packages/admin-ui-docs/content/filter/code.mdx:180]: You can move all those live examples to the overview tab

Comment (anitavincent) [reply packages/admin-ui-docs/content/filter/code.mdx:180]: They're not live yet because I haven't exported the components yet

**Pull request merged**